common/src/main/java/com/microsoft/identity/common/internal/dto/AccountCredentialBase.java

```diff
@@ -52,6 +54,21 @@ public void setAdditionalFields(Map<String, JsonElement> additionalFields) {
         mAdditionalFields = additionalFields;
     }
 
+    /**
+     * Adds the additionalFields of the input AccountCredentialBase to this instance.
+     *
+     * @param other The input AccountCredentialBase to merge with this instance.
+     */
+    public void mergeAdditionalFields(@NonNull final AccountCredentialBase other) {
+        if (null == mAdditionalFields) {
```

Comment:
mAdditionalFields is transient. Wouldn't it be ignored by the serializer?

Reply:
hmm never mind... seems like we're adding these field separately [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/868bb9884d5425691b90b68cbee49d5cd32c465a/common/src/main/java/com/microsoft/identity/common/internal/cache/CacheKeyValueDelegate.java#L125) before serialization (and after deserialization)

Do you know any reason why it's implemented this way?

Reply:
It's implemented this way so that all arbitrary data is written like this:
```json
{
	"arbitrary_prop_1": "value",
	"arbitrary_prop_2": "another_value"
}
```

and not like this:
```json
{
	"mAdditionalFields": {
		"arbitrary_prop_1": "value",
		"arbitrary_prop_2": "another_value"
	}
}
```